lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py

```diff
@@ -0,0 +1,402 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the `fiveg_core_gnb` relation.
+
+This library contains the Requires and Provides classes for handling the `fiveg_core_gnb`
+interface.
+
+The purpose of this library is to provide a way for a FiveG core to provide network information and
+configuration to CUs/gNodeBs.
+
+To get started using the library, you need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.sdcore_nms_k8s.v0.fiveg_core_gnb
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- pydantic
+- pytest-interface-tester
+
+Charms providing the `fiveg_core_gnb` relation should use `FivegCoreGnbProvides`.
+Typical usage of this class would look something like:
+
+    ```python
+    ...
+    from charms.sdcore_gnbsim_k8s.v0.fiveg_core_gnb import FivegCoreGnbProvides
+    ...
+
+    class SomeProviderCharm(CharmBase):
+
+        def __init__(self, *args):
+            ...
+            self.fiveg_core_gnb_provider = FivegCoreGnbProvides(
+                charm=self,
+                relation_name="fiveg_core_gnb"
+                )
+            ...
+            self.framework.observe(
+                self.fiveg_core_gnb_provider.on.fiveg_core_gnb_request,
+                self._on_fiveg_core_gnb_request
+            )
+
+        def _on_fiveg_core_gnb_request(self, event):
+            ...
+            self.fiveg_core_gnb_provider.publish_fiveg_core_gnb_information(
+                relation_id=event.relation_id,
+                tac=tac,
+                plmns=plmns,
+            )
+    ```
+
+    And a corresponding section in charm's `charmcraft.yaml`:
+    ```
+    provides:
+        fiveg_core_gnb:  # Relation name
+            interface: fiveg_core_gnb  # Relation interface
+    ```
+
+Charms that require the `fiveg_core_gnb` relation should use `FivegCoreGnbRequires`.
+Typical usage of this class would look something like:
+
+    ```python
+    ...
+    from charms.sdcore_nms_k8s.v0.fiveg_core_gnb import FivegCoreGnbRequires
+    ...
+
+    class SomeRequirerCharm(CharmBase):
+
+        def __init__(self, *args):
+            ...
+            self.fiveg_core_gnb = FivegCoreGnbRequires(
+                charm=self,
+                relation_name="fiveg_core_gnb"
+            )
+            ...
+            self.framework.observe(self.fiveg_core_gnb.on.gnb_config_available,
```

Comment:
Should we also mention that on `relation-created` event the requirer should push the gNB name to the relation data bag?

Reply:
I guess this question still requires your answer ;)